controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -204,6 +204,7 @@ func (w *Workload) updateCoreDNSDeployment(ctx context.Context, info *coreDNSInf
 	patchCoreDNSDeploymentImage(info.Deployment, info.ToImage)
 	// Flip the deployment volume back to Corefile (from the backup key).
 	patchCoreDNSDeploymentVolume(info.Deployment, corefileBackupKey, corefileKey)
+	// NOTE: This resource isn't managed by the controller. No need to add the WithStatusObservedGeneration option.
```

Comment:
Remove these comments? Seems un-necessary given that patches are all optional

Reply:
I have asked for those comments. this makes it explicit the reasoning behind having the option or not.

Reply:
removed the comments, if you want I can re-add it